In [1]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

#Importe de Datos
df = pd.read_csv("data/Coleccionista.csv")
df.head()

Painting  Price  Bidders  Age
0         1   3470       10   67
1         2   3500        8   56
2         3   3700        7   73
3         4   3860        4   71
4         5   3920       12   99

In [2]:
X = df[['Bidders', 'Age']]
X

Bidders  Age
0        10   67
1         8   56
2         7   73
3         4   71
4        12   99
5        10   87
6        11   78
7         8   83
8        13   90
9        13   98
10        0   91
11        7   93
12        2   97
13        6   79
14        4   83
15        3   71
16        9   89
17        5  103
18        3  106
19        8   93
20        8   88
21        4   96
22        5   94
23        3   88
24        1   84

In [3]:
y =df[['Price']]
y

Price
0    3470
1    3500
2    3700
3    3860
4    3920
5    3900
6    3830
7    3940
8    3880
9    3940
10   4200
11   4060
12   4200
13   4020
14   4190
15   4130
16   4130
17   4370
18   4450
19   4390
20   4380
21   4540
22   4660
23   4710
24   4880

In [4]:
model = LinearRegression()
model.fit(X,y)
y_pred = model.predict(X)

mse = mean_squared_error(y,y_pred)
r2 = r2_score(y,y_pred)

In [5]:
model.intercept_

array([3080.05303581])

In [6]:
print('Coefcientes: ', model.coef_)
print('Error cuadratico de la media', mse)
print('Variaza: ', r2)

Coefcientes:  [[-54.18854049  16.2891028 ]]
Error cuadratico de la media 43396.435062885874
Variaza:  0.647801948911782


In [7]:
Y = model.predict([[10,1000]])
print('El precio de la pintura es: ',Y)

El precio de la pintura es:  [[18827.27043323]]


c:\Users\herna\anaconda3\envs\notebooks\Lib\site-packages\sklearn\utils\validation.py:2749: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Formula

Precio=3080.05−54.19(Postores)+16.29(Antiguedad)

# Relacion de los datos obtenidos

La regresión muestra que la antigüedad incrementa el precio de la pintura, mientras que el número de postores tiene una relación inversa, por lo que mientras mas incrementa el numero de postores el precio disminuye, posiblemente porque las obras más valiosas atraen a menos compradores especializados.